File: backend/app/services/queue_service.py
from app.schemas.queue import QueueSnapshot


_current_queue = QueueSnapshot(areas={"ER": 18, "Lab": 6, "Radiology": 3})


def get_current_queue() -> QueueSnapshot:
    return _current_queue


def update_queue(snapshot: QueueSnapshot) -> QueueSnapshot:
    global _current_queue
    _current_queue = snapshot
    return _current_queue

File: frontend/services/queueService.ts
import { ApiResult, fetchJson } from "@/services/api";
import { QueueSnapshot } from "@/types/queue";

type QueueApiResponse = {
  areas: Record<string, number>;
  captured_at?: string;
};

const fallbackQueue: QueueApiResponse = {
  areas: {
    ER: 18,
    Lab: 6,
    Radiology: 3
  }
};

export async function getQueueSnapshot(): Promise<ApiResult<QueueSnapshot>> {
  const result = await fetchJson<QueueApiResponse>("/queue", fallbackQueue);

  return {
    ...result,
    data: {
      areas: result.data.areas,
      capturedAt: result.data.captured_at
    }
  };
}
